CHANGELOG.md

```diff
@@ -37,6 +37,92 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
    -  When the initials is `undefined`, no gutter space will be reserved for the avatar.
    -  When the initials is `""` (empty string), gutter space will be reserved, but not avatar will be shown.
+-  [`useRenderActivity`](https://github.com/microsoft/BotFramework-WebChat/tree/master/docs/HOOKS.md#userenderactivity) hook is being deprecated, in favor of the new [`useCreateActivityRenderer`](https://github.com/microsoft/BotFramework-WebChat/tree/master/docs/HOOKS.md#usecreateactivityrenderer) hook.
+-  [`useRenderActivityStatus`](https://github.com/microsoft/BotFramework-WebChat/tree/master/docs/HOOKS.md#userenderactivitystatus) hook is being deprecated, in favor of the new [`useCreateActivityStatusRenderer`](https://github.com/microsoft/BotFramework-WebChat/tree/master/docs/HOOKS.md#usecreateactivitystatusrenderer) hook.
+-  [`useRenderAvatar`](https://github.com/microsoft/BotFramework-WebChat/tree/master/docs/HOOKS.md#userenderavatar) hook is being deprecated, in favor of the new [`useCreateAvatarRenderer`](https://github.com/microsoft/BotFramework-WebChat/tree/master/docs/HOOKS.md#usecreateavatarrenderer) hook.
+
+#### Change in general middleware design
+
+> This change will impact middleware which use downstream result.
+
+Previously, when a middleware is called, they are passed with a single argument.
+
+Starting from 4.10.0, multiple arguments could be passed to the middleware. All middleware should pass all arguments to the downstream middleware. This change enables future extension to the middleware pattern.
+
+For example, a passthrough middleware was:
+
+<!-- prettier-ignore-start -->
+```js
+() => next => card => next(card)
+```
+<!-- prettier-ignore-end -->
+
+It should become:
+
+<!-- prettier-ignore-start -->
+```js
+() => next => (...args) => next(...args)
+```
+<!-- prettier-ignore-end -->
+
+This also applies to the render function returned by activity middleware. The previous signature was:
+
+<!-- prettier-ignore-start -->
+```js
+() => next => card => children => next(card)(children)
+```
+<!-- prettier-ignore-end -->
+
+It should become:
+
+<!-- prettier-ignore-start -->
+```js
+() => next => (...setupArgs) => (...renderArgs) => next(...setupArgs)(...renderArgs)
+```
+<!-- prettier-ignore-end -->
+
+> Note: Please read the following section for another change in the activity middleware signature for decorators.
+
+#### Change in activity middleware
+
+> This change will impact activity middleware used for decoration.
+
+Previously, when an activity middleware hide a specific activity from view, it returns a function, `() => false`.
+
+Starting from 4.10.0, if an activity need to be hide from view, the middleware should return `false` instead of `() => false`. This change allows transcript to correctly group activities and ignore activities that are not in view.
+
+To avoid the `TypeError: x is not a function` error, middleware should aware that downstream middleware may return `false` instead of a function.
```

Comment:
```suggestion
To avoid the `TypeError: x is not a function` error, all middleware should be aware that downstream middleware may return `false` instead of a function.
```